### PR TITLE
Add documentation for tracing directive

### PR DESCRIPTION
**Pull request by andriikushch**
**What**

This PR adds documentation for the new module/directive added in this PR:

https://github.com/caddyserver/caddy/pull/4361

### CODE REVIEW

File: src/docs/markdown/caddyfile/directives/tracing.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+title: tracing (Caddyfile directive)
+---
+
+# tracing
+
+It provides integration with OpenTelemetry tracing facilities.
+
+When enabled, it will propagate an existing trace context or initialize a new one.
+
+It is based on `https://github.com/open-telemetry/opentelemetry-go`.
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tracing.md]: Maybe make this a link. (And is that even relevant for the users of the server?)

File: src/docs/markdown/caddyfile/directives/tracing.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+It uses `gRPC` as an exporter protocol and  W3C `tracecontext` and `baggage` as propagators.
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tracing.md]: Possibly link to relevant docs (I don't know what these are, lol)

File: src/docs/markdown/caddyfile/directives/tracing.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+see [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.7.0/specification/protocol/exporter.md).
+
+### Example
+
+```bash
+export OTEL_EXPORTER_OTLP_HEADERS="myAuthHeader=myToken,anotherHeader=value"
+export OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=https://my-otlp-endpoint:55680
+```
+
+### Caddyfile configuration example
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tracing.md]:
```suggestion
## Examples
```

File: src/docs/markdown/caddyfile/directives/tracing.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+### Environment variables
+
+It can be configured using the environment variables defined
+by the [OpenTelemetry Environment Variable Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md).
+
+For the exporter configuration details, please
+see [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.7.0/specification/protocol/exporter.md).
+
+### Example
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tracing.md]:
```suggestion

For example:
```

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks for writing this!

Note that whether we merge this depends on the results of the conversation in https://github.com/caddyserver/caddy/pull/4361. :+1: But either way the docs will be used somewhere if not on our site, like in a README or something.

File: src/docs/markdown/caddyfile/directives/tracing.md (hunk 1 above)
Comment (andriikushch) [reply src/docs/markdown/caddyfile/directives/tracing.md]: Good idea 👍. I made it a link. 

There are a few reasons why I would like to keep it here. 

Most important to me is that `open-telemetry/opentelemetry-go` at the moment does not support some of the environment variables that are used for a configuration (https://github.com/open-telemetry/opentelemetry-go/issues/1698). 

Having this reference here gives some extra information for the webserver users why one or another feature doesn't work.

File: src/docs/markdown/caddyfile/directives/tracing.md (hunk 2 above)
Comment (andriikushch) [reply src/docs/markdown/caddyfile/directives/tracing.md]: I agree 😃  Links added 👍

**Review by mholt — APPROVED** [review]
Thanks for contributing these!

**Pull request merged**